scripts/builtin/shortestPath.dml

```diff
@@ -0,0 +1,157 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+#
+# Computes the minimum distances (shortest-path) between a single 
+# source vertex and every other vertex in the graph.
+# 
+# Documentation; "Pregel: A System for Large-Scale Graph Processing"
+# Grzegorz Malewicz, Matthew H. Austern, Aart J. C. Bilk, 
+# James C. Dehnert, Ikkan Horn, Naty Leiser and Grzegorz Czajkowski
+#
+#
+#
+#----------------------------------
+# *******INPUT PARAMETERS*********
+#----------------------------------
+#
+#	- Adjacency matrix of the labeled graph 
+#	(also considered directed labeled graphs)
+# 	
+#----------------------------------
+# *******OUTPUT PARAMETERS*********
+#----------------------------------
+#	- Matrix of minimum distances (shortest-path) between vertices
+#	(the value of the ith row and the jth column of the output matrix is 
+#	the minimum distance shortest-path from vertex i to vertex j)
+#	- When the value of the minimum distance is -1, the two nodes are not
+#	connected.
+#
+#----------------------------------
+# *******INTERMEDIATE ELEMENTS*********
+#----------------------------------
+#
+#	- neighboursList: matrix (double) of nx3 dimensions, when the minimum 
+#	distance from the source node to a given node is updated,the following
+#	information will be contained in this matrix. Each row represents one 
+#	different update.
+#		- First column: predecesor node which information about the minimum 
+#		distance from the source node has been updated.
+#		- Second column: neighbour that will receive the information in the 
+#		current superstep.
+#		- Third column: updated minimum distance from the source node to the
+#		 node in the first column.
+#	- neighboursListTmp: matrix (double) of nx3 dimensions, during each superstep,
+#	the information about the new updates (not needed until the following superstep)
+#	will be saved in this matrix.
+#
+#
+#
+
+
+s_shortestPath = function(Matrix[Double] G) 
```

Comment:
should take the graph, vertex distances, and vertex id  of the single source id for comparing the shortest paths to all other nodes.